File: macos/Classes/entity/V2TimPermissionGroupInfo.swift
import Foundation
import ImSDKForMac_Plus

/// 自定义消息响应实体
class V2TimPermissionGroupInfo: V2TIMPermissionGroupInfo {

    /// 根据对象获得字典对象
    public static func getDict(info: V2TIMPermissionGroupInfo) -> [String: Any] {
        var result: [String: Any] = [:];
         result["customData"] = info.customData;
         result["groupPermission"] = info.groupPermission;
         result["groupID"] = info.groupID;
         result["permissionGroupID"] = info.permissionGroupID;
         result["permissionGroupName"] = info.permissionGroupName;
         result["memberCount"] = info.memberCount;
         return result;
    }
    init(dict: [String: Any]) {
        super.init();
            
         self.groupID = (dict["groupID"] as? String);
		
		 if let customData = dict["customData"] as? String {
		 	self.customData = customData;
		 }
		 if let groupPermission = dict["groupPermission"] as? UInt64 {
		 	self.groupPermission = groupPermission;
		 }
		 if let permissionGroupID = dict["permissionGroupID"] as? String {
		 	self.permissionGroupID = permissionGroupID;
		 }
		 if let permissionGroupName = dict["permissionGroupName"] as? String {
		 	self.permissionGroupName = permissionGroupName;
		 }
		 
		
    }
}

File: lib/models/v2_tim_permission_group_info.dart
import 'package:tencent_cloud_chat_sdk/utils/utils.dart';

class V2TimPermissionGroupInfo {
  String groupID;
  String permissionGroupID;
  String permissionGroupName;
  String customData;
  int groupPermission;
  int memberCount;
  V2TimPermissionGroupInfo({
    required this.customData,
    required this.groupID,
    required this.groupPermission,
    required this.memberCount,
    required this.permissionGroupID,
    required this.permissionGroupName,
  });
  Map<String, dynamic> toJson() {
    return Map<String, dynamic>.from({
      'groupID': groupID,
      'permissionGroupID': permissionGroupID,
      'permissionGroupName': permissionGroupName,
      'customData': customData,
      'groupPermission': groupPermission,
      'memberCount': memberCount,
    });
  }

  static V2TimPermissionGroupInfo fromJson(Map json) {
    json = Utils.formatJson(json);
    return V2TimPermissionGroupInfo(
      groupID: json['groupID'] ?? "",
      permissionGroupID: json['permissionGroupID'] ?? "",
      permissionGroupName: json['permissionGroupName'] ?? "",
      customData: json['customData'] ?? "",
      groupPermission: json['groupPermission'] ?? 0,
      memberCount: json['memberCount'] ?? 0,
    );
  }
  String toLogString() {
    String res = "groupID:$groupID|permissionGroupID:$permissionGroupID|groupPermission:$groupPermission|memberCount:$memberCount|permissionGroupName:$permissionGroupName";
    return res;
  }
}
